api/synapse-api-rest-imperative/src/main/java/io/americanexpress/synapse/api/rest/imperative/controller/exceptionhandler/ControllerExceptionHandler.java

```diff
@@ -193,4 +194,22 @@ private ResponseEntity<ErrorResponse> handleInternalServerError(final Throwable
 
         return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(errorResponse);
     }
+
+    @ExceptionHandler(OptimisticLockingFailureException.class)
+    public ResponseEntity<ErrorResponse> handleOptimisticLockingFailure(OptimisticLockingFailureException optimisticLockingFailureException) {
+        logger.entry(optimisticLockingFailureException);
+        ResponseEntity<ErrorResponse> errorResponseEntity;
+
+        if (optimisticLockingFailureException.getCause() == null) {
+            ErrorCode errorConflict = ErrorCode.CONFLICT;
+            String message = errorMessagePropertyReader.getErrorMessage(errorConflict);
+            String errorMessage = "The resource you are trying to update is outdated. Please try again.";
+            ErrorResponse errorResponse = new ErrorResponse(errorConflict, errorConflict.getMessage(), errorMessage, message);
+            errorResponseEntity = ResponseEntity.status(errorConflict.getHttpStatus().value()).body(errorResponse);
+        } else {
+            errorResponseEntity = handleInternalServerError(optimisticLockingFailureException);
+        }
```

Comment:
shouldn't we be returning 409 in both cases?

Reply:
I followed the scheme from the other methods in `ControllerExceptionHandler` where the `handleInternalServerError` is called (which returns a 500) if there is no cause received on the exception. If the team agrees we should do 409 in both cases then I can update this. I did the above to make it consistent with what we already have.

Reply:
Yes, here the `OptimisticLockingFailureException` must always be mapped to 409. The cause could possible be an internal db specific exception, nevertheless the `OptimisticLockingFailureException` itself indicates a conflict.

Reply:
Got it. Code updated and pushed